app/controllers/merge_requests_controller.rb

```diff
@@ -22,22 +21,33 @@ def update
     validate_response
 
     if @merge_request.errors.blank? && @merge_request.update(merge_request_params)
+      if page == "merging_organisations"
+        new_merging_org_ids = params["merge_request"]["new_merging_org_ids"].split(" ")
+        new_merging_org_ids.each do |org_id|
+          MergeRequestOrganisation.create!(merge_request: @merge_request, merging_organisation_id: org_id)
+        end
+      end
       redirect_to next_page_path
     else
       render previous_template, status: :unprocessable_entity
     end
   end
 
   def update_merging_organisations
+    @new_merging_org_ids = params["merge_request"]["new_merging_org_ids"].split(" ")
     merge_request_organisation = MergeRequestOrganisation.new(merge_request_organisation_params)
-    if merge_request_organisation.save
+    if merge_request_organisation.valid?
+      @new_merging_org_ids.push(merge_request_organisation_params[:merging_organisation_id])
```

Comment:
We don't want to immediately save added orgs in case user navigates away from the page without saving